ostd/src/mm/page_table/mod.rs

```diff
@@ -97,87 +99,71 @@ impl PageTable<UserMode> {
             self.root.activate();
         }
     }
+}
 
-    /// Clear the page table.
-    ///
-    /// # Safety
-    ///
-    /// The caller must ensure that:
-    ///  1. No other cursors are accessing the page table.
-    ///  2. No other CPUs activates the page table.
-    pub(in crate::mm) unsafe fn clear(&self) {
-        let mut root_node = self.root.clone_shallow().lock();
-        const NR_PTES_PER_NODE: usize = nr_subpage_per_huge::<PagingConsts>();
-        for i in 0..NR_PTES_PER_NODE / 2 {
-            let root_entry = root_node.entry(i);
-            if !root_entry.is_none() {
-                let old = root_entry.replace(Child::None);
-                // Since no others are accessing the old child, dropping it is fine.
-                drop(old);
+impl PageTable<KernelMode> {
+    /// Create a new kernel page table.
+    pub(crate) fn new_kernel_page_table() -> Self {
+        let kpt = Self::empty();
+
+        // Make shared the page tables mapped by the root table in the kernel space.
+        {
+            const NR_PTES_PER_NODE: usize = nr_subpage_per_huge::<PagingConsts>();
+            let kernel_space_range = NR_PTES_PER_NODE / 2..NR_PTES_PER_NODE;
+
+            let guard = crate::task::disable_preempt();
+
+            let mut root_node = kpt.root.borrow().lock(&guard);
+            for i in kernel_space_range {
+                let mut root_entry = root_node.entry(i);
+                let is_tracked = if super::kspace::should_map_as_tracked(
+                    i * page_size::<PagingConsts>(PagingConsts::NR_LEVELS - 1),
+                ) {
+                    MapTrackingStatus::Tracked
+                } else {
+                    MapTrackingStatus::Untracked
+                };
+                let _ = root_entry.alloc_if_none(&guard, is_tracked).unwrap();
             }
         }
+
+        kpt
     }
-}
 
-impl PageTable<KernelMode> {
     /// Create a new user page table.
     ///
     /// This should be the only way to create the user page table, that is to
     /// duplicate the kernel page table with all the kernel mappings shared.
     pub fn create_user_page_table(&self) -> PageTable<UserMode> {
-        let mut root_node = self.root.clone_shallow().lock();
-        let mut new_node =
+        let guard = crate::task::disable_preempt();
```

Comment:
Same as above.
 - `crate::task::disable_preempt();` -> `disable_preempt()`
 - Rearrange the blank lines:
```rust
        let new_root =
            PageTableNode::alloc(PagingConsts::NR_LEVELS, MapTrackingStatus::NotApplicable);

        let guard = crate::task::disable_preempt();
        let mut root_node = self.root.borrow().lock(&guard);
        let mut new_node = new_root.borrow().lock(&guard);

        // ...
```